PA193_mnemonic_Slytherin/test_mnemonic.py

```diff
@@ -207,12 +207,25 @@ def test_generate(self):
             self.assertEqual(Mnemonic(test_vector[1]), mnemonic)
             self.assertEqual(Seed(unhexlify(test_vector[2])), seed)
 
+    def test_generate_invalid_password_too_long(self):
+        entropy = Entropy(unhexlify('00000000000000000000000000000000'))
+        password = 'a' * 1024 * 1024 * 1024 * 2  # 2 GB
+        with self.assertRaises(ValueError):
+            generate(entropy, password)
+
     def test_recover(self):
         for test_vector in TREZOR_TEST_VECTORS['english']:
             entropy, seed = recover(Mnemonic(test_vector[1]), TREZOR_PASSWORD)
             self.assertEqual(Entropy(unhexlify(test_vector[0])), entropy)
             self.assertEqual(Seed(unhexlify(test_vector[2])), seed)
 
+    def test_recover_invalid_password_too_long(self):
+        mnemonic = Mnemonic('abandon abandon abandon abandon abandon abandon'
+                            ' abandon abandon abandon abandon abandon about')
+        password = 'a' * 1024 * 1024 * 1024 * 2  # 2 GB
```

Comment:
here I get 'OverflowError: repeated string is too long' its probably not needed to test for this long passwords

Reply:
Yep, 1GB would be enough:)

Reply:
i think this might depend on the platform, but 1 GB is also valid for me

Reply:
Yes, it would depend on specific machine... Purpose of these tests is that when such long string gets into standard PBKDF2, it crashes. So we had to limit password length. I can make it shorter so that we do not test whether PBKDF2 crashes, but do test whether our own implementation of PBKDF2 raises `ValueError`.

Reply:
fixed in e0099ea